src/radical/pilot/agent/scheduler/continuous_fifo.py

```diff
@@ -19,40 +22,103 @@ class ContinuousFifo(Continuous):
     #
     def __init__(self, cfg, session):
 
-        self.slots = None
-        self._last = -1   # nothing has run, yet - next up is unit.000000
-
         Continuous.__init__(self, cfg, session)
 
-        # we want to check all units in the wait pool on open slots
-        self._uniform_waitpool = False
-
 
     # --------------------------------------------------------------------------
     #
-    def _try_allocation(self, cu):
+    def _configure(self):
+
+        Continuous._configure(self)
+
+        self._last              = -1   # nothing has run, yet
+        self._ordered_wait_pool = list()
+        self._ordered_wait_lock = mt.RLock()  # look on the above set
+
+
+    # --------------------------------------------------------------------------
+    # overload the main method from the base class
+    def work(self, units):
+
+        if not isinstance(units, list):
+            units = [units]
+
+        self.advance(units, rps.AGENT_SCHEDULING, publish=True, push=False)
+
+        # add incoming units to the wait list, sort it again by unit ID
+        with self._ordered_wait_lock:
+
+            # cache ID int to avoid repeated parsing
+            for unit in units:
+                if 'serial_id' not in unit:
+                    unit['serial_id'] = int(unit['uid'].split('.')[-1])
+
+                assert(unit['serial_id'] > self._last)
+
+            # add units to pool and resort it
+            self._ordered_wait_pool += units
+            self._ordered_wait_pool.sort(key=lambda x: x['serial_id'])
```

Comment:
Isn't 'units' already a list (hence ordered)? Is this required in that case? Since the 1) new units (which will be ordered) will have a serial id greater than those in the wait pool and 2) you append the new units to the end of the wait pool.

Reply:
Nvm, I think I understand why this is required for the wait pool.

Reply:
Hey Vivek - this PR is outdated - I should remove it.  Please have a look at #1704 which implements a more generic ordering schema, which does not depend on the unit IDs.